Find a functionally equivalent code:android.media.AudioRecord.audioBuffSizeCheck:COMMENT
Method Modifier: private     
Comment:// mNativeBufferSizeInBytes is valid (multiple of frame size, positive)

@@ -1,7 +1,7 @@
 {
     // NB: this section is only valid with PCM data.
     // To update when supporting compressed formats
-    int frameSizeInBytes = mChannelCount * (mAudioFormat == AudioFormat.ENCODING_PCM_8BIT ? 1 : 2);
+    int frameSizeInBytes = mChannelCount * (AudioFormat.getBytesPerSample(mAudioFormat));
     if ((audioBufferSize % frameSizeInBytes != 0) || (audioBufferSize < 1)) {
         throw new IllegalArgumentException("Invalid audio buffer size.");
     }

